src/lib/coda_base/snark_transition.ml

```diff
@@ -80,13 +81,53 @@ module Make (Inputs : Inputs_intf) :
     ; ledger_proof: Proof.t option }
   [@@deriving bin_io, sexp, fields]
 
-  type value =
-    ( Blockchain_state.value
-    , Consensus_data.value
-    , Sok_message.Digest.Stable.V1.t
-    , Currency.Amount.t )
-    t
-  [@@deriving bin_io, sexp]
+  module Value = struct
+    module Stable = struct
+      module V1 = struct
+        module T = struct
+          let version = 1
+
+          (* "deriving" incompatible with "nonrec", break definition loop *)
+          type ( 'blockchain_state
+               , 'consensus_data
+               , 'sok_digest
+               , 'supply_increase ) tt =
+            ( 'blockchain_state
+            , 'consensus_data
+            , 'sok_digest
+            , 'supply_increase )
+            t
+          [@@deriving bin_io, sexp]
+
+          type t =
+            ( Blockchain_state.value
+            , Consensus_data.value
+            , Sok_message.Digest.Stable.V1.t
+            , Currency.Amount.t )
```

Comment:
These functored-over types also need to versioned, right? Should we change the functor interface?

Reply:
I'm confused on how to handle this issue, see my remarks on Discord.